### PR TITLE
Add nlp tip on compact transformers

### PR DESCRIPTION
**Pull request by TDehaene**
New NLP tip on the usage and benefit on compact transformers (sub 25mio parameters).

Main take-away seems to be that implementation is crucial: the ones I had the highest hopes for (albert and mobilebert) don't seem to fare that well, while the baby-berts seem to do rather well.

Distilbert stills looks like a great all-round model, but in case you have extra restrictions, the investigated models definitely seems to have value

### CODE REVIEW

**Review by MatsUy — COMMENTED** [review]
Nice tip, really interesting :)
Also curious as to the reason why there are large differences in training and inference times.
I wanted to suggest that it would also be nice to compare it with a big BERT model, but since this notebook already takes quite some time, it's probably better not to.

File: nlp/2021_02_26_compact_transformers/compact_transformers.ipynb [hunk 1]
```diff
@@ -0,0 +1,3861 @@
...
+      "cell_type": "code",
+      "metadata": {
+        "id": "76Pu02HM7AI3"
+      },
+      "source": [
+        "for model in run_dict:\r\n",
+        "    test_pipe = pipeline(\"sentiment-analysis\",f\"tests/{model.get('model_name')}\", device=-1)\r\n",
+        "\r\n",
+        "    start_time = time.time()\r\n",
+        "    for i in range(100):\r\n",
```
[the remark this answers is not in the document]
Comment (TDehaene) [reply nlp/2021_02_26_compact_transformers/compact_transformers.ipynb:3343]: That's a good point, I'll check it out how if fairs against batched predictions

File: nlp/2021_02_26_compact_transformers/compact_transformers.ipynb [hunk 2]
```diff
@@ -0,0 +1,3861 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4Hp3Msndwc3y"
+      },
+      "source": [
+        "## Compact Transformers  🤏\n",
+        "\n",
+        "We all know bigger is often better! Nothing cooler than rolling up in your larger-than-life Transformer model. Who doesn't want to drive a monster truck once in their life...\n",
+        "\n",
+        "However, monster truck aren't always practical in real life. Models need to run smoothly in production. So the best model size is often the one most suited for your practical situation and requirements.\n",
```
[the remark this answers is not in the document]
Comment (TDehaene) [reply nlp/2021_02_26_compact_transformers/compact_transformers.ipynb:2732]: 🚚 > 🚚🚚🚚

File: nlp/2021_02_26_compact_transformers/compact_transformers.ipynb [hunk 3]
```diff
@@ -0,0 +1,3861 @@
...
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "-rcrjcbtxqgB"
+      },
+      "source": [
+        "### Setup"
```
Comment (ThomasVrancken) [inline nlp/2021_02_26_compact_transformers/compact_transformers.ipynb:2760]: Could maybe be nice to put a `nvidia-smi` here, to check whether there's a GPU.

**Review by ThomasVrancken — COMMENTED** [review]
Very nice TOTW! Super instructiv, great format and I really like the thoroughness in evaluating different aspects (cool metrics & plots). Also glad to see that DistilBERT performs well.

File: nlp/2021_02_26_compact_transformers/compact_transformers.ipynb [hunk 4]
```diff
@@ -0,0 +1,3861 @@
...
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 542
+        },
+        "id": "AC-8prB_vKwD",
+        "outputId": "cc83710a-b133-44f5-b401-3a2b7875f2ab"
+      },
+      "source": [
+        "import plotly.express as px\r\n",
```
[the remark this answers is not in the document]
Comment (TDehaene) [reply nlp/2021_02_26_compact_transformers/compact_transformers.ipynb:3581]: Indeed, thought about mpl for a bit, but figured that most people will follow the colab notebook link, where the graphs are already visible (stored state) interactively

File: nlp/2021_02_26_compact_transformers/compact_transformers.ipynb (hunk 3 above)
Comment (TDehaene) [reply nlp/2021_02_26_compact_transformers/compact_transformers.ipynb:2760]: That's a good point, will add

File: nlp/2021_02_26_compact_transformers/compact_transformers.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (TDehaene) [reply nlp/2021_02_26_compact_transformers/compact_transformers.ipynb]: @MatsUy : added a new test that, for each model:

- determines the max batch size without going OOM
- does a prediction with this batch size
- thus calculates a theoretical max predictions/s, taking into account the max batchsize
- and plots this in a graph

**Review by ThomasVrancken — APPROVED** [review]

**Pull request merged**